pkg/build/build.go

```diff
@@ -777,6 +779,49 @@ func (ctx *Context) OverlayBinSh() error {
 func (ctx *Context) PopulateCache() error {
 	ctx.Logger.Printf("populating cache from %s", ctx.CacheDir)
 
+	// --cache-dir=gs://bucket/path/to/cache first pulls all found objects to a
+	// tmp dir which is subsequently used as the cache.
+	if strings.HasPrefix(ctx.CacheDir, "gs://") {
+		cctx := context.TODO()
+
+		tmp, err := os.MkdirTemp("", "melange-cache")
+		if err != nil {
+			return err
+		}
+		bucket, prefix, _ := strings.Cut(strings.TrimPrefix(ctx.CacheDir, "gs://"), "/")
+
+		client, err := storage.NewClient(cctx)
+		if err != nil {
+			return err
+		}
+		b := client.Bucket(bucket)
+		it := b.Objects(cctx, &storage.Query{Prefix: prefix})
+		for {
+			attrs, err := it.Next()
+			if err != nil {
+				return fmt.Errorf("failed to get next remote cache object: %w", err)
+			}
+			on := attrs.Name
+			rc, err := b.Object(on).NewReader(cctx)
+			if err != nil {
+				return fmt.Errorf("failed to get reader for next remote cache object %s: %w", on, err)
+			}
+			w, err := os.Create(filepath.Join(tmp, on))
```

Comment:
I wonder if it would be helpful to log each object it pulls from the cache.  That way we can be certain that it is still doing things?

Reply:
Sure, done.